ui/server/src/test/scala/pl/touk/nussknacker/ui/api/ProcessesResourcesSpec.scala

```diff
@@ -160,6 +177,66 @@ class ProcessesResourcesSpec extends FunSuite with ScalatestRouteTest with Match
     }
   }
 
+  test("search processes by categories") {
+    val firstProcessor = ProcessName("Processor1")
+    val secondProcessor = ProcessName("Processor2")
+
+    createProcess(firstProcessor, testCategoryName, false)
+    createProcess(secondProcessor, secondTestCategoryName, false)
+
+    Get(s"/processes") ~> routWithAllPermissions ~> check {
+      status shouldEqual StatusCodes.OK
+      val data = responseAs[String].decodeOption[List[BasicProcess]].get
+      data.size shouldBe 2
+    }
+
+    Get(s"/processes?categories=$testCategoryName") ~> routWithAllPermissions ~> check {
+      status shouldEqual StatusCodes.OK
+      val data = responseAs[String].decodeOption[List[BasicProcess]].get
+      data.size shouldBe 1
+    }
+
+    Get(s"/processes?categories=$secondTestCategoryName") ~> routWithAllPermissions ~> check {
+      status shouldEqual StatusCodes.OK
+      val data = responseAs[String].decodeOption[List[BasicProcess]].get
+      data.size shouldBe 1
+    }
+
+    Get(s"/processes?categories=$secondTestCategoryName,$testCategoryName") ~> routWithAllPermissions ~> check {
+      status shouldEqual StatusCodes.OK
+      val data = responseAs[String].decodeOption[List[BasicProcess]].get
+      data.size shouldBe 2
+    }
+  }
+
+  test("search processes by isDeployed") {
+    val firstProcessor = ProcessName("Processor1")
+    val secondProcessor = ProcessName("Processor2")
+    val thirdProcessor = ProcessName("Processor3")
+
+    createProcess(firstProcessor, testCategoryName, false)
+    createProcess(secondProcessor, testCategoryName, false)
+    createDeployedProcess(thirdProcessor, testCategoryName, false)
+
+    Get(s"/processes") ~> routWithAllPermissions ~> check {
+      status shouldEqual StatusCodes.OK
+      val data = responseAs[String].decodeOption[List[BasicProcess]].get
+      data.size shouldBe 3
+    }
+
+    Get(s"/processes?isDeployed=true") ~> routWithAllPermissions ~> check {
```

Comment:
maybe we can add also assertion for ids ;)

Reply:
Good Idea :)